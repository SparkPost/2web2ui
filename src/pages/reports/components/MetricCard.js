import React from 'react';
import PropTypes from 'prop-types';
import { InfoOutline } from '@sparkpost/matchbox-icons';
import useHibanaOverride from 'src/hooks/useHibanaOverride';
import useUniqueId from 'src/hooks/useUniqueId';
import { Panel, Tooltip } from 'src/components/matchbox';
import { Heading } from 'src/components/text';
import OGStyles from './MetricCard.module.scss';
import hibanaStyles from './MetricCardHibana.module.scss';

const MetricCard = ({ label, value, tooltipContent, ...rest }) => {
  const styles = useHibanaOverride(OGStyles, hibanaStyles);
  const tooltipId = useUniqueId('metric-card');
  const tooltip = tooltipContent ? (
    <Tooltip id={tooltipId} dark content={tooltipContent} horizontalOffset="-1.1rem">
      <InfoOutline className={styles.TooltipIcon} />
    </Tooltip>
  ) : null;

  return (
    <div className={styles.MetricCard}>
      <Panel.LEGACY className={styles.Panel} {...rest}>
        <Panel.LEGACY.Section className={styles.PanelSection}>
          <Heading as="h3" className={styles.Value}>
            {value}
          </Heading>
          <Heading as="h4" className={styles.Label}>
            {label} {tooltip}
          </Heading>
        </Panel.LEGACY.Section>
      </Panel.LEGACY>
    </div>
  );
};

MetricCard.propTypes = {
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.node]),
  tooltipContent: PropTypes.string,
  label: PropTypes.string,
};

export default MetricCard;
