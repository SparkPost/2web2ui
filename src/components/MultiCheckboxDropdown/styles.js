import { Button } from 'src/components/matchbox';
import { ChevronRight } from '@sparkpost/matchbox-icons';
import styled from 'styled-components';

export const StatusPopoverContent = styled.span`
  display: inline-block; /* Necessary to supply width & cut ellipse joined string */
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;

  margin-right: ${props =>
    props.theme.space['400']}; /* because we AlignedButtonIcon absolute right */

  > * {
    /* Hacky fix, but addresses vertical centering without introducing a flex parent that wreaks havoc on text truncation */
    display: inline-block;
    transform: translateY(2px);
  }
`;

export const AlignedTextButton = styled(Button)`
  text-align: left;
  line-height: ${props =>
    props.theme.space[
      '450'
    ]}; /* Note: Fixes text and button alignment so they're all the same vertical alignment (see styles/note on span children) */

  > span,
  > span > span {
    vertical-align: middle; /* Note: Fixes text and button alignment so they're all the same vertical alignment (v align the children of the button) */
  }

  width: 100%;
`;

export const AlignedButtonIcon = styled(Button.Icon)`
  position: absolute;
  right: ${props => props.theme.space['300']}; /* Note: same as Matchbox listbox svg icon */
`;

export const Chevron = styled(ChevronRight)`
  color: ${props => props.theme.colors.blue['700']};
  transform: rotate(90deg);
`;
