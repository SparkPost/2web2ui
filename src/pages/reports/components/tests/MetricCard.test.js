import React from 'react';
import { shallow } from 'enzyme';
import useHibanaOverride from 'src/hooks/useHibanaOverride';
import MetricCard from '../MetricCard';
import styles from '../MetricCard.module.scss';

jest.mock('src/hooks/useHibanaOverride');

const props = {
  label: 'Card Label',
  value: 'Card Value',
  tooltipContent: 'Tooltip Content',
  primary: true,
};

describe('MetricCard Component: ', () => {
  beforeEach(() => {
    useHibanaOverride.mockReturnValue(styles);
  });

  it('should render correctly with all props', () => {
    const wrapper = shallow(<MetricCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render without tooltip', () => {
    const wrapper = shallow(<MetricCard {...props} />);
    wrapper.setProps({ tooltipContent: null });
    expect(wrapper).toMatchSnapshot();
  });
});
