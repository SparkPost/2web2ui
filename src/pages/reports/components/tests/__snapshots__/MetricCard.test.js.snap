// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetricCard Component:  should render correctly with all props 1`] = `
<div
  className="MetricCard"
>
  <Panel.LEGACY
    className="Panel"
    primary={true}
  >
    <Panel.LEGACY.Section
      className="PanelSection"
    >
      <Heading
        as="h3"
        className="Value"
      >
        Card Value
      </Heading>
      <Heading
        as="h4"
        className="Label"
      >
        Card Label
         
        <Tooltip
          content="Tooltip Content"
          dark={true}
          horizontalOffset="-1.1rem"
          id="metric-card-1"
        >
          <InfoOutline
            className="TooltipIcon"
          />
        </Tooltip>
      </Heading>
    </Panel.LEGACY.Section>
  </Panel.LEGACY>
</div>
`;

exports[`MetricCard Component:  should render without tooltip 1`] = `
<div
  className="MetricCard"
>
  <Panel.LEGACY
    className="Panel"
    primary={true}
  >
    <Panel.LEGACY.Section
      className="PanelSection"
    >
      <Heading
        as="h3"
        className="Value"
      >
        Card Value
      </Heading>
      <Heading
        as="h4"
        className="Label"
      >
        Card Label
         
      </Heading>
    </Panel.LEGACY.Section>
  </Panel.LEGACY>
</div>
`;
